packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.props.ts

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import {
+    ManagedClasses,
+    ViewportPositionerClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-base";
+
+export interface ViewportPositionerManagedClasses
+    extends ManagedClasses<ViewportPositionerClassNameContract> {}
+export interface ViewportPositionerUnhandledProps
+    extends React.HTMLAttributes<HTMLDivElement> {}
+
+export enum ViewportPositionerHorizontalPosition {
+    left = "left",
+    centerLeft = "centerLeft",
+    centerRight = "centerRight",
+    right = "right",
+    uncontrolled = "uncontrolled",
+}
+
+export enum ViewportPositionerVerticalPosition {
+    top = "top",
+    centerTop = "centerTop",
+    centerBottom = "centerBottom",
+    bottom = "bottom",
+    uncontrolled = "uncontrolled",
+}
+
+export enum AxisPositioningMode {
+    uncontrolled = "uncontrolled",
+    flipOutward = "flipOutward",
+    flipInward = "flipInward",
+}
+
+export interface ViewportPositionerHandledProps extends ViewportPositionerManagedClasses {
+    /**
+     *  Ref to the element the positioner is anchored to
+     */
+    anchor?: React.RefObject<any>;
+
+    /**
+     *  Ref to the viewport the positioner is constrained to
+     */
+    viewport?: React.RefObject<any>;
+
+    /**
+     *  The positioning mode for the horizontal axis, default is uncontrolled
```

Comment:
Does this mean that it won't position itself by default?

Reply:
Yes, that is the current configuration as I had difficulty identifying what default should be.  Thoughts?

Reply:
uhhh maybe bottom left? I don't know that it matters too much, but I would expect to see it do *something* by default - otherwise it just appears broken on first run without configuration.

Reply:
How about leaving the horizontal access uncontrolled and the vertical axis to flipOutwards.  That would be a vanilla setting for a vertical menu.

Reply:
did that.